httpx/utils.py

```diff
@@ -212,6 +213,28 @@ def kv_format(**kwargs: typing.Any) -> str:
     return " ".join(f"{key}={value!r}" for key, value in kwargs.items())
 
 
+def should_not_be_proxied(url: "URL") -> bool:
+    """ Return True if url should not be proxied,
+    return False otherwise.
+    """
+    no_proxy = getproxies().get("no")
+    if not no_proxy:
+        return False
+    no_proxy_list = [host.strip() for host in no_proxy.split(",")]
+    for name in no_proxy_list:
+        if name == "*":
+            return True
+        if name:
+            name = name.lstrip(".")  # ignore leading dots
+            name = re.escape(name)
+            pattern = r"(.+\.)?%s$" % name
```

Comment:
Nit: I might be missing something, but do we absolutely need to use a regex here? This pattern seems like it doing `endswith`?

Reply:
it does nearly just `endswith`, if it contains data before name (the `.+`) then it has to have a `.` before name, so whole domains match (see test with example2.com matching against ample2.com, it think)